src/DeviceModeling/device_modeler.h

```diff
@@ -1077,11 +1077,12 @@ class device_modeler {
                   << '\n';
       }
     }
-    block->instance_vector().push_back(std::make_shared<device_block_instance>(
-        block, block->instance_vector().size(), logic_location_x_i,
-        logic_location_y_i, logic_address_i, name, io_bank,
-        logic_location_z_i));
-    block->instances()[name] = block->instance_vector().back();
+    parent_block->instance_vector().push_back(
+        std::make_shared<device_block_instance>(
+            block, block->instance_vector().size(), logic_location_x_i,
+            logic_location_y_i, logic_address_i, name, io_bank,
+            logic_location_z_i));
+    parent_block->instances()[name] = block->instance_vector().back();
```

Comment:
You should not just directly overwrite the instance_map_ (or instances()) without giving any error to user.

There is function you should call add_instance() - https://github.com/os-fpga/FOEDAG/blob/main/src/DeviceModeling/device_block.h#L482-L487
- it did the checking
- it throw error if the infra is misused

Reply:
Beside the suggested add_instance(), there are more issue in these changes. It will cause invalid memory access.

I will create new PR with proper fix, plus other issues (more on parameter)